File: Camera/Camera/SelfieTableViewCell.m
//
//  SelfieTableViewCell.m
//  Camera
//
//  Created by Joe E. on 10/19/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

#import "SelfieTableViewCell.h"

@interface SelfieTableViewCell ()

@property (weak, nonatomic) IBOutlet UIImageView *selfieImageView;

@end

@implementation SelfieTableViewCell

//set image based on selfie["image"]

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

}

- (void) setSelfie:(PFObject *)selfie {
    _selfie = selfie;
    [self updateInfo];

}

- (void)updateInfo {
    
    PFFile *userImageFile = self.selfie[@"image"];
    NSLog(@"userImageFile is working");
    [userImageFile getDataInBackgroundWithBlock:^(NSData * imageData, NSError * _Nullable error) {
        if (!error) {
            UIImage * image = [UIImage imageWithData:imageData];
        }
        
    }];
    
//    self.selfieImageView.image = self.selfie[@"image"];
}

@end

File: Selfie/SelfieTableViewCell.swift
//
//  SelfieTableViewCell.swift
//  Selfie
//
//  Created by Joe E. on 11/1/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

import UIKit
import Parse

//Strings
private let IMAGE = "image"

class SelfieTableViewCell: UITableViewCell {
    
    var selfie = PFObject()
    
    func selectSelfie(selfie:PFObject) {
        self.selfie = selfie
        self.updateInfo()
        
    }
    
    func updateInfo() {
        let userImageFile = selfie[IMAGE]
        NSLog("userImageFileIsWorking")
        userImageFile.getDataInBackgroundWithBlock { (imageData:NSData?, error:NSError?) -> Void in
            if error == nil {
                let image = UIImage(data: imageData!)
            }
        }
    
    
    }

    override func awakeFromNib() {
        super.awakeFromNib()

    }

    override func setSelected(selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

}
